lib/core/security/tokenRepository.js

```diff
@@ -393,13 +394,16 @@ class TokenRepository extends Repository {
       debug('Loading API keys into Redis');
 
       const promises = [];
-      const createToken = async ({token, ttl, userId }) => {
-        return await this.persistForUser(token, userId, ttl);
+      const createToken = async apiKey => {
+        return await this.persistForUser(
+          apiKey.authToken(),
+          apiKey._source.userId,
+          apiKey._source.ttl);
```

Comment:
Why not include the token prefixes when you store it?
And in all cases, in fact.
You could isolate the prefix removal in the token verification code (with a `@deprecated` check to skip prefix removal if there isn't one, to ensure backward-compatibility).

I'm asking because I worry about potential inconsistencies, where in some parts of the code the tokens don't have prefixes, and where they do in some other parts. It can be hard to follow and to debug.

Especially with the fingerprint that seems to be computed w/ the prefix, whereas the token is stored without one.

Reply:
Very good point :+1: